### PR TITLE
Telemetry: target.proto

### PR DESCRIPTION
**Pull request by sepehrfrgh**
This PR adds observation gotags into `target.proto` and `target_service.proto` for telemetry purposes

### CODE REVIEW

File: internal/proto/controller/api/resources/accounts/v1/account.proto [hunk 1]
```diff
@@ -41,23 +41,23 @@ message Account {
...
+  google.protobuf.Timestamp created_time = 50 [json_name = "created_time"]; // @gotags: `class:"public" eventstream:"observation"`
 
   // Output only. The time this resource was last updated.
-  google.protobuf.Timestamp updated_time = 60 [json_name = "updated_time"]; // @gotags: `class:"public"`
+  google.protobuf.Timestamp updated_time = 60 [json_name = "updated_time"]; // @gotags: `class:"public" eventstream:"observation"`
 
   // Version is used in mutation requests, after the initial creation, to ensure this resource has not changed.
   // The mutation will fail if the version does not match the latest known good version.
-  uint32 version = 70; // @gotags: `class:"public"`
+  uint32 version = 70; // @gotags: `class:"public" eventstream:"observation"`
```
Comment (johanbrandhorst) [inline internal/proto/controller/api/resources/accounts/v1/account.proto:51]: I've seen us skip the version in other PRs (and in other protos in this PR), is this correct?

File: internal/proto/controller/api/services/v1/account_service.proto [hunk 2]
```diff
@@ -136,16 +136,16 @@ message UpdateAccountResponse {
...
 
 message DeleteAccountResponse {}
 
 message SetPasswordRequest {
-  string id = 1; // @gotags: `class:"public"`
+  string id = 1; // @gotags: `class:"public" eventstream:"observation"`
   // Version is used to ensure this resource has not changed.
   // The mutation will fail if the version does not match the latest known good version.
-  uint32 version = 2; // @gotags: `class:"public"`
+  uint32 version = 2; // @gotags: `class:"public" eventstream:"observation"`
```
Comment (johanbrandhorst) [inline internal/proto/controller/api/services/v1/account_service.proto:148]: Should this be captured?

File: internal/proto/controller/api/services/v1/account_service.proto [hunk 3]
```diff
@@ -154,10 +154,10 @@ message SetPasswordResponse {
 }
 
 message ChangePasswordRequest {
-  string id = 1; // @gotags: `class:"public"`
+  string id = 1; // @gotags: `class:"public" eventstream:"observation"`
   // Version is used to ensure this resource has not changed.
   // The mutation will fail if the version does not match the latest known good version.
-  uint32 version = 2; // @gotags: `class:"public"`
+  uint32 version = 2; // @gotags: `class:"public" eventstream:"observation"`
```
Comment (johanbrandhorst) [inline internal/proto/controller/api/services/v1/account_service.proto:160]: Ditto

File: internal/proto/controller/api/resources/accounts/v1/account.proto (hunk 1 above)
Comment (sepehrfrgh) [reply internal/proto/controller/api/resources/accounts/v1/account.proto:51]: Based on my understanding this is not a product-type version and it is only being used for mutation so I did not include them in the previous proto and PR and I think I accidentally added here 😞

Comment (sepehrfrgh) [reply internal/proto/controller/api/resources/accounts/v1/account.proto:51]: @jimlambrt Please let me know what you thing about his version and adding to observation

Comment (jimlambrt) [reply internal/proto/controller/api/resources/accounts/v1/account.proto:51]: I don't think we need the resource versions.  I think, I've seen them included in other PRs.

Comment (jimlambrt) [reply internal/proto/controller/api/resources/accounts/v1/account.proto:51]: If we're not going to emit them in the observation I see some re-PRs of previous protos in our future.

File: internal/proto/controller/api/resources/accounts/v1/account.proto [hunk 4]
```diff
@@ -83,10 +83,10 @@ message Account {
   }
 
   // Output only. managed_group_ids indicates IDs of the managed groups that currently contain this account
-  repeated string managed_group_ids = 110 [json_name = "managed_group_ids"]; // @gotags: `class:"public"`
+  repeated string managed_group_ids = 110 [json_name = "managed_group_ids"]; // @gotags: `class:"public" eventstream:"observation"`
 
   // Output only. The available actions on this resource for this user.
-  repeated string authorized_actions = 300 [json_name = "authorized_actions"]; // @gotags: `class:"public"`
+  repeated string authorized_actions = 300 [json_name = "authorized_actions"]; // @gotags: `class:"public" eventstream:"observation"`
```
Comment (jimlambrt) [inline internal/proto/controller/api/resources/accounts/v1/account.proto:89]: Hum... do we need this?

File: internal/proto/controller/api/resources/authmethods/v1/auth_method.proto [hunk 5]
```diff
@@ -82,10 +82,10 @@ message AuthMethod {
 
   // Output only. Whether this auth method is the primary auth method for it's scope.
   // To change this value update the primary_auth_method_id field on the scope.
-  bool is_primary = 110 [json_name = "is_primary"]; // @gotags: `class:"public"`
+  bool is_primary = 110 [json_name = "is_primary"]; // @gotags: `class:"public" eventstream:"observation"`
 
   // Output only. The available actions on this resource for this user.
-  repeated string authorized_actions = 300 [json_name = "authorized_actions"]; // @gotags: `class:"public"`
+  repeated string authorized_actions = 300 [json_name = "authorized_actions"]; // @gotags: `class:"public" eventstream:"observation"`
```
Comment (jimlambrt) [inline internal/proto/controller/api/resources/authmethods/v1/auth_method.proto:88]: do we need to know the authorized actions for the requesting user?

File: internal/proto/controller/api/resources/groups/v1/group.proto [hunk 6]
```diff
@@ -50,21 +50,21 @@ message Group {
...
   uint32 version = 80; // @gotags: `class:"public"`
 
   // Output only. Contains the list of member IDs in this Group.
-  repeated string member_ids = 90 [json_name = "member_ids"]; // @gotags: `class:"public"`
+  repeated string member_ids = 90 [json_name = "member_ids"]; // @gotags: `class:"public" eventstream:"observation"`
 
   // Output only. The members of this Group.
   repeated Member members = 100;
 
   // Output only. The available actions on this resource for this user.
```
Comment (jimlambrt) [inline internal/proto/controller/api/resources/groups/v1/group.proto:68]: same question

File: internal/proto/controller/api/resources/hostcatalogs/v1/host_catalog.proto [hunk 7]
```diff
@@ -80,7 +80,7 @@ message HostCatalog {
   string secrets_hmac = 120; // @gotags: `class:"public"`
 
   // Output only. The available actions on this resource for this user.
-  repeated string authorized_actions = 300 [json_name = "authorized_actions"]; // @gotags: `class:"public"`
+  repeated string authorized_actions = 300 [json_name = "authorized_actions"]; // @gotags: `class:"public" eventstream:"observation"`
```
Comment (jimlambrt) [inline internal/proto/controller/api/resources/hostcatalogs/v1/host_catalog.proto:83]: same question.

File: internal/proto/controller/api/resources/targets/v1/target.proto [hunk 8]
```diff
@@ -212,7 +212,7 @@ message Target {
   }
 
   // Output only. The available actions on this resource for this user.
-  repeated string authorized_actions = 300 [json_name = "authorized_actions"]; // @gotags: `class:"public"`
+  repeated string authorized_actions = 300 [json_name = "authorized_actions"]; // @gotags: `class:"public" eventstream:"observation"`
```
Comment (jimlambrt) [inline internal/proto/controller/api/resources/targets/v1/target.proto:215]: same question.

File: internal/proto/controller/api/services/v1/auth_method_service.proto [hunk 9]
```diff
@@ -96,15 +96,15 @@ service AuthMethodService {
...
+  string id = 1; // @gotags: `class:"public" eventstream:"observation"`
 }
 
 message GetAuthMethodResponse {
   resources.authmethods.v1.AuthMethod item = 1;
 }
 
 message ListAuthMethodsRequest {
-  string scope_id = 1 [json_name = "scope_id"]; // @gotags: `class:"public"`
+  string scope_id = 1 [json_name = "scope_id"]; // @gotags: `class:"public" eventstream:"observation"`
```
Comment (jimlambrt) [inline internal/proto/controller/api/services/v1/auth_method_service.proto:107]: do we want recursive (below)?

**Review by jimlambrt — CHANGES_REQUESTED** [review]

**Review by sepehrfrgh — COMMENTED** [review]
FYI: I had to rebase with `llb-telemetry` and that would get rid of those duplicate commits. I am glad that happen so we found out we shouldn't include the `autorized_action`, `version`, and `recursive` flags on rpc list services

File: internal/proto/controller/api/resources/targets/v1/target.proto [hunk 10]
```diff
@@ -297,7 +297,7 @@ message SshTargetAttributes {
       this: "attributes.enable_session_recording"
```
Comment (johanbrandhorst) [inline internal/proto/controller/api/resources/targets/v1/target.proto:297]: Should we not include the storage bucket id?

Comment (sepehrfrgh) [reply internal/proto/controller/api/resources/targets/v1/target.proto:297]: No, these new features are going to be for phase 2 of the telemetry

Comment (johanbrandhorst) [reply internal/proto/controller/api/resources/targets/v1/target.proto:297]: Why is enable_session_recording set then?

**Review by johanbrandhorst — APPROVED** [review]

**Review by jimlambrt — APPROVED** [review]

**Pull request merged**